### ISSUE TITLE
[A11Y] Focus state should be thicker

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#5277]
### 8. Focus state should be thicker

**Description of Issue**: Based on work in progress WCAG guidelines, we recommend making the focus state 3-4px thick to make it visible for folks with low vision.
**URL**: https://pr3750-llnmetezlaix5oe10ddq5m3nya7zdz2b.tugboat.qa/
**Testing Environment/Browser**: Windows Lenovo PC, Windows 10 Pro, Chrome Browser v. 94.0.4606.61, JAWS 2020.12.13
**[Severity](https://github.com/backdrop/backdrop-issues/issues/5244#issuecomment-931739260)**: P1
**Solution**: Most of the site currently features a light black focus outline on interactive elements (see screenshots below for details). At times, like on the main navigation, the links gets a white outline. 

In upcoming (work-in-progress) WCAG guidelines, it is also recommended that focus state is at least 2px wide, or "The focus indication area is greater than or equal to the longest side of the bounding rectangle of the focused control, times 2 CSS pixels."  I recommend that the focus state be 3px or 4px wide.

Also, some of the focus around the links are getting cut off. Please investigate and try to fix this.

Resource for work-in-progress WCAG specification on focus indication:
https://www.w3.org/WAI/WCAG21/Understanding/focus-visible-enhanced 

**Screenshot:**
![image](https://user-images.githubusercontent.com/12848168/135542120-46eac921-0c64-4731-b2a2-c4cf7d532c68.png)
![image](https://user-images.githubusercontent.com/12848168/135542179-80c07848-1da9-411b-a337-37c1baef54ab.png)
![image](https://user-images.githubusercontent.com/12848168/135542276-c90888d2-cd86-408d-b03f-02505e841c51.png)

**Comment by stpaultim** [comment]
We worked on this as a group as part of Open Source day. Our theory is that the following css should address this problem:

```
*:focus-visible {
  outline-style: solid;
  outline-width: 4px;
}
```
We tried this in this file modules/system/css/system.theme.css and it did not work. 

Just for fun, we tried the same thing in basis (see current Pull Request) and it seems to work. 

Clearly, the solution to this problem can not be Basis specific. Does anyone have any idea how we can get this or a better solution to work in system css.

**Comment by stpaultim** [comment]
This was before:

![before](https://user-images.githubusercontent.com/3144571/135705988-c93ff8d6-f2cd-4a2b-ab65-0c7858c1c473.png)

This is what the PR (in Basis) does:

![after](https://user-images.githubusercontent.com/3144571/135706011-06be37bd-9fca-4f97-92f7-04411f2287e5.png)

Is this the correct css improvement?
How do we get this to work in core (not basis)?
Are the issues with backwards compatibility?

**Comment by stpaultim** [comment]
@klonos - So from your comment and change in the issue title, you seem to be of the opinion that the correct place to fix this is in the themes, not in core? 

Wouldn't it make sense to have a default width provided by core - for all the folks that won't think to theme this in their own custom themes? 

We started out trying to fix this in core. The PR is only in Basis, because we couldn't get it to work properly in core.

**Comment by philsward** [comment]
Dropping this here for additional reading and thoughts.

I for one, didn't know :focus-visible was for keyboard use. Good to know. That possibly means :focus should be used for mouse and :focus-visible should be used for keyboard/screen reader navigation. This may need additional research. If this is true, then I suggest a style for :focus when a mouse has clicked a box for focus and a different style for keyboard navigation. They are two completely different UX situations in my mind and totally CAN be different. The residing accessibility expert might want to weigh in on this though.

Safari doesn't currently appear to recognize :focus-visible however it [appears they are working on it](https://blogs.igalia.com/mrego/2021/06/07/focus-visible-in-webkit-may-2021/)

Some good general info about :focus-visible
https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible

@stpaultim You might try adding the CSS as a single line instead of each one broken out:

```
:focus-visible {
  outline: dashed 3px orange;
}
```

I added this to system.theme.css and it seems to work fine when viewing on Seven (because it's part of the "system"). Basis on the other hand, doesn't even have this styling set at all, so it's falling back to the _user agent stylesheet_

![image](https://user-images.githubusercontent.com/3436110/136318381-ed91598d-f15b-4ead-9ae1-865139ce1a6a.png)
